pkgs/by-name/ar/artalk/package.nix

```diff
@@ -31,20 +33,35 @@ buildGoModule rec {
     "-X github.com/ArtalkJS/Artalk/internal/config.Version=${version}"
     "-X github.com/ArtalkJS/Artalk/internal/config.CommitHash=${version}"
   ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
   preBuild = ''
     tar -xzf ${web}
     cp -r ./artalk_ui/* ./public
   '';
 
+  postInstall =
+    ''
+      ln -s $out/bin/Artalk $out/bin/artalk
```

Comment:
Doing this will add 2 binary in the output BIN path. Is it really what you're looking for? Basically, you're not only renaming, but merely adding a new bin in its path.

Reply:
yeah...`mv` may more suitable in this case
https://github.com/NixOS/nixpkgs/pull/315892#discussion_r1636075104
I accepted the suggestion before because of the combability

Reply:
finally I choose `mv`.
with `lib.getExe` and `meta.mainProgram`, I believe this is compatible enough.